app/assets/javascripts/application/multiple-files.js

```diff
@@ -0,0 +1,211 @@
+mumuki.load(() => {
+  class File {
+    static get NAME_CLASS() { return '.file-name'; }
+    static get DELETE_BUTTON_CLASS() { return '.delete-file-button'; }
+
+    constructor(tab, editor) {
+      this.tab = tab;
+      this.editor = editor;
+    }
+
+    initialize(name) {
+      this.name = name;
+      this.unselect();
+
+      return this;
+    }
+
+    get name() {
+      return this.tab.find(File.NAME_CLASS).text();
+    }
+
+    set name(name) {
+      return this.tab.find(File.NAME_CLASS).text(name);
+    }
+
+    get isSelected() {
+      return this.tab.hasClass("active");
+    }
+
+    setUpOnRemove(handler) {
+      this.tab.find(File.DELETE_BUTTON_CLASS).click(() => {
+        handler(this);
+      });
+    }
+
+    remove() {
+      const wasSelected = this.isSelected;
+
+      this.tab.remove();
+      this.editor.remove();
+
+      return wasSelected;
+    }
+
+    select() {
+      this._selectElement(this.tab);
+      this._selectElement(this.editor);
+    }
+
+    unselect() {
+      this._unselectElement(this.tab);
+      this._unselectElement(this.editor);
+    }
+
+    _selectElement(element) {
+      element.addClass('active');
+      element.addClass('in');
+    }
+
+    _unselectElement(element) {
+      element.removeClass('active');
+      element.removeClass('in');
+    }
+  }
+
+  class MultipleFileEditor {
+    constructor(tabsContainer, editorsContainer) {
+      this.tabsContainer = tabsContainer;
+      this.editorsContainer = editorsContainer;
+
+      this.MAX_TABS = 5;
+
+      this._addFileButton = this.tabsContainer.siblings('.add-file-button');
+      this._updateButtonsVisibility();
+    }
+
+    get files() {
+      const editors = this.editors;
+
+      return this.tabs.map((i, tab) => {
+        return new File($(tab), $(editors[i]));
+      });
+    }
+
+    get tabs() {
+      return this.tabsContainer.children();
+    }
+
+    get editors() {
+      return this.editorsContainer.find('.file-editor')
+    }
+
+    setUpAddFile() {
+      this._addFileButton.click(() => {
+        this._addFile();
+      });
+    }
+
+    setUpDeleteFiles() {
+      this.files.each((i, file) => {
+        this.setUpDeleteFile(file);
+      });
+    }
+
+    setUpDeleteFile(file) {
+      file.setUpOnRemove(this._deleteFile.bind(this));
+    }
+
+    _addFile() {
+      const name = prompt('Insert a file name'); // TODO: i18n, or improve somehow
```

Comment:
I'll leave the original @flbulgarelli's comment here:
> Oops! @fedescarpa what can we do here? It is very important that the UI is fully internationalized


I added i18n on this message putting the [locale](https://github.com/mumuki/mumuki-laboratory/blob/36847ceecc4589200a83e36e4c44e641e13e9d6b/app/helpers/multiple_file_editor_helper.rb#L7) on a hidden [input](https://github.com/mumuki/mumuki-laboratory/blob/36847ceecc4589200a83e36e4c44e641e13e9d6b/app/views/layouts/exercise_inputs/editors/_multiple_files.html.erb#L16), but I think we need a better solution. This won't be a problem when we use a proper bootstrap's modal dialog